coderd/coderd.go

```diff
@@ -245,9 +249,10 @@ func New(options *Options) *API {
 			Authorizer: options.Authorizer,
 			Logger:     options.Logger,
 		},
-		metricsCache: metricsCache,
-		Auditor:      atomic.Pointer[audit.Auditor]{},
-		Experiments:  experiments,
+		metricsCache:          metricsCache,
+		Auditor:               atomic.Pointer[audit.Auditor]{},
+		TemplateScheduleStore: atomic.Pointer[provisionerdserver.TemplateScheduleStore]{},
```

Comment:
If we decide to detach provisionerdserver from coderd, this strong dependency would be a problem. Do you think that there is an alternative solution possible?

Reply:
What strong dependency? Do you mean the import dependency? Coderd uses this interface as part of the template update route to do the Set call.

I'm assuming moving this interface to a separate package solves this concern if you mean import dependency

Reply:
Yes, the idea is to not increase coupling between packages. If we keep binding modules, then it will harder to untangle it.